### DISCUSSION TITLE
What's your torch version?

### DISCUSSION THREAD

**Discussion by JeonghwaYoo-R** [open-edge-platform/anomalib#1165] [Q&A] [answered]
I want to run the latest version of anomalib. (anomalib-1.0.0.dev0)

However, this error is occured. 

> ImportError: cannot import name 'KeypointRCNN_KeypointRCNN_ResNet50_FPN_Weights' from 'torchvision.models.detection'

I think it's because torchvision doesn't match your environment.

But I couldn't find which version of torch you guys used.

Which torch version did you use?

**Comment by samet-akcay** [comment]
hmm, this might mean that the name of the weights might have changed in one of the recent torchvision versions. @djdameln, do you think if this is related to AI-VAD?

**Reply by JeonghwaYoo-R** [reply]
@samet-akcay 

Thank you for your reply.
Yes! The error is occured in `anomalib/models/ai_vad/features.py`

I ran with Torch 1.11.0 version and 2.0.0 version. And the error occurred in both versions.

**Reply by JeonghwaYoo-R** [reply]
with Torch 2.0.0 

The code in the import part in `anomalib/models/ai_vad/features.py` should be changed.

from 
`from torchvision.models.detection import KeypointRCNN_ResNet50_FPN_Weights`

to

`from torchvision.models.detection.keypoint_rcnn import KeypointRCNN_ResNet50_FPN_Weights`

I solved it by changing this, but I still want to know which version of torch you used.

**Reply by djdameln** [reply, marked as answer]
On my side, `from torchvision.models.detection import KeypointRCNN_ResNet50_FPN_Weights` works fine with both
```
torch                     1.13.1+cu116
torchvision               0.14.1+cu116
```
and 
```
torch                     2.0.1
torchvision               0.15.2
```

**Comment by samet-akcay** [comment]
<img width="636" alt="image" src="https://github.com/openvinotoolkit/anomalib/assets/17574420/d9def053-1e02-4ea3-84fc-1f5b1d677f72">

It also works on my side

**Comment by samet-akcay** [comment]
@JeonghwaYoo-R, what is your torchvision version?

**Reply by JeonghwaYoo-R** [reply]
@samet-akcay 

My torchvision version is 0.15.0 and torch version is 2.0.0. 


Strangely, currently no import error occurs in the code `from torchvision.models.detection import KeypointRCNN_ResNet50_FPN_Weights`

And in the `torchvision.models.detection code`,  importing everything in `keypoint_rcnn`. 

Thus I wonder why `from torchvision.models.detection import KeypointRCNN_ResNet50_FPN_Weights` gives an error and `from torchvision.models.detection.keypoint_rcnn import KeypointRCNN_ResNet50_FPN_Weights` doesn't.


I think in the process of constantly reinstalling torch, it seems that something happened in my environment for a while.


There is no problem with your code currently. So I will close this discussion. thanks for the reply!
